src/brogue/Items.c

```diff
@@ -1629,6 +1617,50 @@ void itemName(item *theItem, char *root, boolean includeDetails, boolean include
     return;
 }
 
+void itemKindName(item *theItem, char *kindName) {
+
+    // use lookup table for randomly generated items with more than one kind per category
+    if (theItem->category & (ARMOR | CHARM | FOOD | POTION | RING | SCROLL | STAFF | WAND | WEAPON)) {
+        strcpy(kindName, tableForItemCategory(theItem->category, NULL)[theItem->kind].name);
+    } else {
+        switch (theItem->category) {
+            case KEY:
+                strcpy(kindName, keyTable[theItem->kind].name); //keys are not randomly generated but have a lookup table
+                break;
+            // these items have only one kind per category and no lookup table
+            case GOLD:
+                strcpy(kindName, "gold peices");
+                break;
+            case AMULET:
+                strcpy(kindName, "amulet of yendor");
+                break;
+            case GEM:
+                strcpy(kindName, "lumenstone");
+                break;
+            default:
+                strcpy(kindName, "unknown");
+                break;
+        }
+    }
+}
+
+void itemRunicName(item *theItem, char *runicName) {
+    char vorpalEnemyMonsterClass[15] ="";
+
+    strcpy(runicName, "\0"); //assume no runic
+
+    if (theItem->flags & ITEM_RUNIC) {
+        if (theItem->flags & ITEM_RUNIC_VORPAL) {
```

Comment:
Is it necessary to have this flag, couldn't we just check if the runic was W_SLAYING or A_IMMUNITY?

Reply:
Ah, well no, it's not strictly necessary, but it does save something like 5 whole lines of code. A bit of overzealous abstraction on my part I suppose. I'll refactor it.

Reply:
Done. As it turns out, I was in fact saving exactly zero lines of code. lol.